src/app/dataset-block/metadata-block/metadata-block.component.ts

```diff
@@ -58,9 +69,30 @@ export class MetadataBlockComponent
         );
     }
 
+    public onPageChange(params: {
+        currentPage: number;
+        isClick: boolean;
+    }): void {
+        this.loadHistory(params.currentPage - 1);
```

Comment:
Unlike `ngOnInit`, there is no subscription registration here, so I guess there is a subscription leak here.

Reply:
Added trackSubscription method here